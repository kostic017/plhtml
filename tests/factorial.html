<!doctype html>
<html lang="en">
    <head>
        <title>`Factorial`</title>
    </head>
    <body>

        <main>

            <var class="integer">quit</var>

            <div data-while="quit &ne; 1">

                <var class="integer">n</var>
                <var class="integer">i</var>
                <var class="integer">fact</var>

                <data value="1">i</data>
                <data value="1">fact</data>

                <output>`n: `</output>
                <input name="n">

                <div data-while="i &leq; n">
                    <data value="fact * i">fact</data>
                    <data value="i + 1">i</data>
                </div>

                <output>`Factorial of ` + n + ` is ` + fact</output>

                <output>`\nType 1 to quit: `</output>
                <input name="quit">

            </div>

        </main>

    </body>
</html>
