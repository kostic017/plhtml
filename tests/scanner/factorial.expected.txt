<
!
Doctype
Html
>
<
Html
Lang
=
"
Identifier|en|
"
>
<
Head
>
<
Title
>
StringConst|Factorial|
<
/
Title
>
<
/
Head
>
<
Body
>
<
Main
>
<
Var
Class
=
"
Identifier|integer|
"
>
Identifier|quit|
<
/
Var
>
<
Div
Data
-
While
=
"
Identifier|quit|
NeqOp
IntConst|1|
"
>
<
Var
Class
=
"
Identifier|integer|
"
>
Identifier|n|
<
/
Var
>
<
Var
Class
=
"
Identifier|integer|
"
>
Identifier|i|
<
/
Var
>
<
Var
Class
=
"
Identifier|integer|
"
>
Identifier|fact|
<
/
Var
>
<
Data
Value
=
"
IntConst|1|
"
>
Identifier|i|
<
/
Data
>
<
Data
Value
=
"
IntConst|1|
"
>
Identifier|fact|
<
/
Data
>
<
Output
>
StringConst|n: |
<
/
Output
>
<
Input
Name
=
"
Identifier|n|
"
>
<
Div
Data
-
While
=
"
Identifier|i|
LeqOp
Identifier|n|
"
>
<
Data
Value
=
"
Identifier|fact|
*
Identifier|i|
"
>
Identifier|fact|
<
/
Data
>
<
Data
Value
=
"
Identifier|i|
+
IntConst|1|
"
>
Identifier|i|
<
/
Data
>
<
/
Div
>
<
Output
>
StringConst|Factorial of |
+
Identifier|n|
+
StringConst| is |
+
Identifier|fact|
<
/
Output
>
<
Output
>
StringConst|\nType 1 to quit: |
<
/
Output
>
<
Input
Name
=
"
Identifier|quit|
"
>
<
/
Div
>
<
/
Main
>
<
/
Body
>
<
/
Html
>
